docs/generate_extension_cat_db.py

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env python3
+
+import json
+import sys
+
+
+def main():
+  extension_db = sys.argv[1]
+  extension_cat_db = sys.argv[2]
+
+  with open(extension_db) as f:
+    data = json.load(f)
+
+  categories = {}
+  for k, v in data.items():
+    for cat in v['categories']:
+      categories[cat] = categories.get(cat, [])
```

Comment:
Suggest using a `defaultdict` here to simplify.

Reply:
i shifted this code elsewhere, and slightly reformulated...